### PR TITLE
Developer instructions

### PR DESCRIPTION
**Pull request by lmoureaux**
## Description

(On top of #123)

Add instructions for developers to the Sphinx documentation:

* Duplicate the user setup instructions
* Duplicate the developer instructions from `.github/CONTRIBUTING.md`
* Add developer setup instructions and documentation guidelines

The concerns raised in #121 are also addressed in this PR.

Since Github has basic support for rendering `rst`, I could add a link to developer docs in the README. However, cross-references don't work, and the rendering is much worse than what Sphinx does.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

We need to document how to set up a development area and test changes.

## How Has This Been Tested?

The instructions are based on https://gist.github.com/jsturdy/4f9bccbdd18d953f73777e492a71fd24, my own experience as a developer and support I did privately on Mattermost.

### Screenshots (if appropriate)

Better than a screenshot, [a link to the documentation](http://homepage.iihe.ac.be/~lmoureau/GEM/sphinx/html/index.html)!

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by bdorney** [comment]
Does this supersede https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/123 or require https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/123?

File: doc/coding.rst [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+        git submodule update
+
+    This will update your local version of `gembuild`_.
+
+Once the package is built, it can be installed using:
+
+.. code-block:: bash
+
+    pip install -U rpm/gempython_gemplotting-1.0.0.tar.gz
```
Comment (bdorney) [inline doc/coding.rst]: The developer `gist` has this done with option `-I` to ignore and force re-install.  In some cases I've found that the `-U` does not always overwrite correctly previously installed files.  Would suggest to have `-I`.

File: doc/coding.rst [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+We have been utilizing a `very helpful guideline`_ for our development model.
+The basic idea is the following:
+
+* Fork from ``cms-gem-daq-project/gem-plotting-tools``
+* Create a branch to develop your particular feature (based off of ``develop``,
+  or in some cases, the current release branch)
+
+  * ``hotfix`` may be created from ``master``
+  * Once that feature is completed, create a pull request
```
Comment (bdorney) [inline doc/coding.rst]: We should also apply the same hotfix to develop.  Add a line here.

**Review by bdorney — CHANGES_REQUESTED** [review]
two small changes needed. Otherwise very good.  Some clarification of which PR's this needs/replaces required.

**Comment by bdorney** [comment]
Does this also superseded: https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/121?

**Comment by lmoureaux** [comment]
No PR is superseded. This one (#125) is built on top of #123 since I expect #123 to be merged before (but it's not required and I can rebase off current `develop`). It also fixes the issue that prevents #121 from being merged (#121 will need a rebase).

**Comment by lmoureaux** [comment]
* Rebased on top of `develop`
* Made the requested changes

**Comment by bdorney** [comment]
This PR is out of date with develop.  Can you update it?  Then I'll approve it.

**Comment by lmoureaux** [comment]
Rebased.

**Review by bdorney — APPROVED** [review]

**Review by mexanick — APPROVED** [review]

**Pull request merged**